### PR TITLE
exclaim-ui onChange action

### PR DESCRIPTION
**Pull request by deverstalmage**
This PR introduces a new property on the `exclaim-ui` component: `onChange`. `onChange` should be a function, and will be called when any bound value in the environment changes. The function will receive an argument in the form `{ key, value }` where `key` is the changed key in the environment.

The `Environment` class now extends `Ember.Evented` to support `change` events that are triggered by bound values changing.

I've also changed the `__metaFor__` property on the `Envrionment` to `__resolveMeta__` to make it clearer. 

@dfreeman

### CODE REVIEW

File: addon/-private/environment/index.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ export default class Environment extends EmberObject {
   setUnknownProperty(key, value) {
     createEnvComputed(this, key, `__bound__.${findIndex(this.__bound__, key)}`);
     set(this, key, value);
+    this.trigger('change', { key, value });
```
Comment (dfreeman) [inline addon/-private/environment/index.js]: The `set` on the line above should already trigger this change, shouldn't it?

File: addon/-private/environment/data.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const {
  * Wraps an object, resolving any Bindings in it when requested to the corresponding
  * paths in the given environment.
  */
-export default class EnvironmentData extends EmberObject {
+export default class EnvironmentData extends EmberObject.extend(Ember.Evented) {
```
Comment (dfreeman) [inline addon/-private/environment/data.js]: Should `EnvironmentData` need to be evented?

File: tests/integration/components/exclaim-ui-test.js [hunk 3]
```diff
@@ -116,6 +117,32 @@ test('it writes bound data back to the env', function(assert) {
...
+  set(this, 'env', {
+    envValue: 'before'
+  });
+
+  this.renderUI();
+
+  run(() => fillIn('input', 'after'))
+
+  assert.ok(get(this, 'onChange').calledOnce);
+  assert.ok(get(this, 'onChange').calledWith({ key: 'envValue', value: 'after' }));
```
Comment (dfreeman) [inline tests/integration/components/exclaim-ui-test.js]: I wonder if we actually need to send the value up? I could imagine the signature just being `calledWith('envValue')`

File: addon/-private/environment/data.js (hunk 2 above)
Comment (deverstalmage) [reply addon/-private/environment/data.js]: I thought it would but turns out no

File: addon/-private/environment/index.js (hunk 1 above)
Comment (deverstalmage) [reply addon/-private/environment/index.js]: it does- I'm surprised this didn't trigger the `onChange` action twice

File: tests/integration/components/exclaim-ui-test.js (hunk 3 above)
Comment (deverstalmage) [reply tests/integration/components/exclaim-ui-test.js]: I suppose not, you could just get the value off of the environment since you'll have the key

**Pull request merged**